src/main/java/core/basesyntax/services/impl/Buy.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.services.ActionInterface;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Buy implements ActionInterface {
+
+    @Override
+    public void action(Storage storage, Fruit fruit) {
+        List<Fruit> list = storage.getFruits().stream()
+                .filter(storageFruit -> storageFruit.getType().equals(fruit.getType()))
+                .filter(storagefruit -> storagefruit.getDate().isAfter(fruit.getDate())
+                        || storagefruit.getDate().isEqual(fruit.getDate()))
+                .sorted()
+                .collect(Collectors.toList());
+
+        int summ = list.stream()
+                .mapToInt(Fruit::getStock_balance)
+                .sum();
+
+        if (summ < fruit.getStock_balance()) {
+            throw new RuntimeException("Недостаточно товаров на складе");
+        } else {
+            list.forEach(fruit1 -> {
```

Comment:
avoid complex bodies for functional interfaces in Stream API

Reply:
fixed it